### PR TITLE
Chore: Use dskit/spanlogger

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Use `dskit/spanlogger`.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: pkg/logql/engine.go [hunk 1]
```diff
@@ -76,7 +79,11 @@ func NewEngine(opts EngineOpts, q Querier, l Limits) *Engine {
 
 // Query creates a new LogQL query. Instant/Range type is derived from the parameters.
 func (ng *Engine) Query(params Params) Query {
+	if ng.logger == nil {
+		panic("doof")
+	}
```
Comment (owen-d) [inline pkg/logql/engine.go:84]:
```suggestion
```

File: pkg/logql/engine.go [hunk 2]
```diff
@@ -59,15 +60,17 @@ func (opts *EngineOpts) applyDefault() {
...
+	logger    log.Logger
 	timeout   time.Duration
 	evaluator Evaluator
 	limits    Limits
 }
 
 // NewEngine creates a new LogQL Engine.
-func NewEngine(opts EngineOpts, q Querier, l Limits) *Engine {
+func NewEngine(opts EngineOpts, q Querier, l Limits, logger log.Logger) *Engine {
 	opts.applyDefault()
```
Comment (owen-d) [inline pkg/logql/engine.go:71]:
```suggestion
	opts.applyDefault()
	if logger == nil {
		logger = log.NewNopLogger()
	}
```

File: pkg/logql/engine.go [hunk 3]
```diff
@@ -105,7 +113,10 @@ type query struct {
 
 // Exec Implements `Query`. It handles instrumentation & defers to Eval.
 func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
-	log, ctx := spanlogger.New(ctx, "query.Exec")
+	if q.logger == nil {
+		panic("oof")
+	}
```
Comment (owen-d) [inline pkg/logql/engine.go:118]:
```suggestion
```

**Review by owen-d — COMMENTED** [review]
Thanks for this PR; I'm sure it wasn't a fun one to put together but I appreciate it. I've left a few nits to avoid panics, then LGTM

File: pkg/util/log/log.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+var (
+	// Logger is a shared go-kit logger.
+	// TODO: Change all components to take a non-global logger via their constructors.
+	// Prefer accepting a non-global logger as an argument.
+	Logger = log.NewNopLogger()
+)
```
Comment (jeschkies) [inline pkg/util/log/log.go:16]: Where is this logger still used? Should we rather address the TODO?

Comment (aknuds1) [reply pkg/util/log/log.go:16]: @jeschkies I hope to be able to return to this soon :/

**Comment by aknuds1** [comment]
Closing in favour of #5187.

**Pull request closed without merging**